source/developers/development_environment.markdown

```diff
@@ -11,20 +11,55 @@ footer: true
 
 You'll need to set up a development environment if you want to develop a new feature or component for Home Assistant. Read on to learn how to set up.
 
-* Visit the [Home Assistant repository](https://github.com/home-assistant/home-assistant) and click **Fork**.
-
-* Consider setting up a virtual environment using [`venv`](https://docs.python.org/3.4/library/venv.html) before running the setup script.
+### {% linkable_title Setup Local Repository %}
 
+Visit the [Home Assistant repository](https://github.com/home-assistant/home-assistant) and click **Fork**.
+Once forked, setup your local copy of the source using the commands:
 ```bash
 $ git clone https://github.com/YOUR_GIT_USERNAME/home-assistant.git
 $ cd home-assistant
 $ git remote add upstream https://github.com/home-assistant/home-assistant.git
+```
+
+### {% linkable_title Prepare Your Environment %}
+
+#### {% linkable_title Core dependencies %} 
+
+Install the core dependencies.
+```bash
+$ sudo apt-get install python3-pip python3-dev libssl-dev
+```
+
+#### {% linkable_title Frontend dependencies (optional) %} 
+
+In order to run or develop the [Frontend](https://home-assistant.io/developers/frontend/) you will need node.js. The preferred method of installing node is [nvm](https://github.com/creationix/nvm). Install nvm using the instructions in the [README](https://github.com/creationix/nvm#install-script), and install node.js by running the following command:
+```bash
+$ nvm install node
+```
+
+#### {% linkable_title Setting up virtual environment (optional) %} 
+
+If you plan on providing isolation to your environment using [`venv`](https://docs.python.org/3.4/library/venv.html). You'll need to install the python3-venv package:
+```bash
+$ sudo apt-get install python3-venv
+```
+Within the `home-assistant` directory, create and activate your virtual environment using the commands:
+```bash
+$ pyvenv venv
```

Comment:
When you say automatically pick up the venv, are you suggesting you don't need to run the activate script? As just a quick test doesn't show this. regardless of if I setup the venv in a sub directory or at the top level, I still need to run activate in order for pip list or pip3 list to show only the modules installed in the virtual environment. 

Also it is recommended to install that the top level, then it may also need changes to .gitignore to help avoid parts of the venv being included in submissions.

I'm happy to make further changes, just need a definitive agreement, or I if you can point me at some public docs that talk about the reason for setting it up in the top level, perhaps we can include both options.

Note, in my latest submit I've also included the fact that pyvenv is being dreprecated and replaced by python3 -m venv.